cmd/service-controller/definition_monitor.go

```diff
@@ -224,8 +225,15 @@ func (m *DefinitionMonitor) getServiceDefinitionFromAnnotatedService(service *co
 				Name:     service.ObjectMeta.Name,
 				Selector: utils.StringifySelector(service.Spec.Selector),
 			}
-			if targetPort := deduceTargetPortFromService(service); targetPort != 0 {
-				target.TargetPort = targetPort
+			if curSvc, ok := m.annotated[svc.Address]; !ok {
```

Comment:
I *think* what we want to do here is check whether there is an originalTargetPort annotation, and if so use that as the targetPort on the target?

Reply:
Yes, now that we have it in the service, I think I can change it to validate based on the new annotation.